lib-js/interactor.ts

```diff
@@ -0,0 +1,198 @@
+import readline from "readline"
+import TestCase from "./test-case"
+
+// Select properties of the test case that can be used in requirements
+export type TestCaseArg = Pick<TestCase, "impl" | "actual" | "result">
```

Comment:
Test typing — this way we can test the functions involved without having to create a whole test case and running it. I admit it's kind of a hack, I'll see if I can come up with a better idea. Probably an interface that the TestCase type implements, instead of doing it backwards like this?